### PR TITLE
Add Close Tab Button

### PR DESCRIPTION
**Pull request by TimChen44**
Provide a tab closure event. After defining the event, the tab displays a closure button, and the control itself does not perform any logic. The closure function is implemented by the user themselves.
tab

### CODE REVIEW

File: Radzen.Blazor/RadzenTabs.razor.cs [hunk 1]
```diff
@@ -71,6 +71,13 @@ public partial class RadzenTabs : RadzenComponent
 
         List<RadzenTabsItem> tabs = new List<RadzenTabsItem>();
 
+        /// <summary>
+        /// Gets or sets tab close button event 
+        /// </summary>
+        /// <value>The close button callback.</value>
+        [Parameter]
+        public EventCallback<RadzenTabsItem> TabCloseClick { get; set; }
```
Comment (akorchev) [inline Radzen.Blazor/RadzenTabs.razor.cs:79]: Please rename it to just `TabClose`

File: Radzen.Blazor/RadzenTabsItem.razor.cs [hunk 2]
```diff
@@ -138,6 +138,14 @@ async Task OnClick()
             }
         }
 
+        async Task OnTabCloseClick()
+        {
+            if (Tabs.TabCloseClick.HasDelegate)
```
Comment (akorchev) [inline Radzen.Blazor/RadzenTabsItem.razor.cs:143]: There is no need for the if check here.

File: Radzen.Blazor/RadzenTabsItem.razor [hunk 3]
```diff
@@ -16,6 +16,14 @@
             else
             {
                 <span class="rz-tabview-title">@Text</span>
+                @if (Tabs.TabCloseClick.HasDelegate)
+                {
+                    <span class="rz-ml-05">
+                        <RadzenButton Click="@OnTabCloseClick" class="rz-border-radius" Icon="close" Variant="Variant.Text"
+                                      ButtonStyle="ButtonStyle.Dark" Shade="Shade.Darker" Size="ButtonSize.ExtraSmall"
```
Comment (akorchev) [inline Radzen.Blazor/RadzenTabsItem.razor:23]: Having both ButtonStyle.Dark and Shade.Darker may look odd in a dark theme. What if those are not set at all?

**Comment by TimChen44** [comment]
Fixed the above issue and added the close button display mode.
The FREE style has been modified. I don't know how to modify the PRO style, so I didn't make any changes.

GIF 2023-9-13 22-43-26

**Comment by akorchev** [comment]
We discussed this internally and decided we should implement something else - a Template property of RadzenTabsItem which would allow adding arbitrary content - buttons, badge etc. The template would display inside the `<a>` element. Would you like to try implementing that?

**Comment by TimChen44** [comment]
The common scenarios here are the close button and badge, but there may be other buttons in some complex scenarios. Supporting templates is a good idea, and I can try to improve it.

**Comment by akorchev** [comment]
It turns out there is a Template property already 🤣 Perhaps we can just make a demo which implements the closing with buttons that are defined in the template. Also the common UX pattern is to display the close button on hover only.

**Comment by TimChen44** [comment]
One problem with existing templates is that users need to create everything themselves, which increases the complexity of user usage for commonly used scenarios.

**Comment by TimChen44** [comment]
Added BadgeText attribute, I feel the effect is quite good.

**Comment by akorchev** [comment]
Hi @TimChen44,

Thank you for your efforts however we decided to not continue with this PR. We prefer to use the Template for such purposes.

**Pull request closed without merging**